### PR TITLE
Add new opcode POPITEM

### PR DESCRIPTION
**Pull request by erikzhang**
Close #386

### CODE REVIEW

File: src/neo-vm/ExecutionEngine.cs [hunk 1]
```diff
@@ -1202,6 +1202,14 @@ private void ExecuteInstruction()
                         x.Clear();
                         break;
                     }
+                case OpCode.POP:
```
Comment (shargon) [inline src/neo-vm/ExecutionEngine.cs:1205]: `POPITEM`? in order to avoid confusion

**Comment by erikzhang** [comment]
@spencercorwin Do you think it solve the problem?

File: src/neo-vm/ExecutionEngine.cs (hunk 1 above)
Comment (erikzhang) [reply src/neo-vm/ExecutionEngine.cs:1205]: What confusion?

Comment (shargon) [reply src/neo-vm/ExecutionEngine.cs:1205]: With the method `Pop()`

File: src/neo-vm/ExecutionEngine.cs [hunk 2]
```diff
@@ -1202,6 +1202,14 @@ private void ExecuteInstruction()
...
+                    {
+                        VMArray x = Pop<VMArray>();
+                        int index = x.Count - 1;
```
Comment (shargon) [inline src/neo-vm/ExecutionEngine.cs:1208]:
```suggestion
                        int index = x.Count - 1;
                        if (index < 0) throw new IndexOutOfRangeException();
```

Comment (ixje) [reply src/neo-vm/ExecutionEngine.cs:1208]: what is index exceeds length of `x`

Comment (erikzhang) [reply src/neo-vm/ExecutionEngine.cs:1208]: `RemoveAt()` will throw the same exception.

**Review by vncoelho — APPROVED** [review]
If it solves the problems it looks good to me.

**Comment by spencercorwin** [comment]
That should work. Let me test it today and confirm that it works.

**Comment by spencercorwin** [comment]
Looks like this works really well! Thank you @erikzhang and team for the quick help.

**Review by shargon — DISMISSED** [review]

File: src/neo-vm/ExecutionEngine.cs (hunk 1 above)
Comment (erikzhang) [reply src/neo-vm/ExecutionEngine.cs]: Updated.

**Review by shargon — APPROVED** [review]

**Pull request merged**